_data-prepper/pipelines/configuration/sources/s3.md

```diff
@@ -10,18 +10,18 @@ nav_order: 20
 
 ## Overview
 
-This is a source plugin that reads events from [Amazon Simple Storage Service](https://aws.amazon.com/s3/) (Amazon S3) objects.
+`s3` is a source plugin that reads events from [Amazon Simple Storage Service](https://aws.amazon.com/s3/) (Amazon S3) objects.
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
 notification_type | Yes | String | Must be `sqs`
-compression | No | String | The compression algorithm to apply: `none`, `gzip`, or `automatic`. Default is `none`.
-codec | Yes | Codec | The codec to apply. Must be `newline`, `json`, or `csv`.
+compression | No | String | The compression algorithm to apply: `none`, `gzip`, or `automatic`. Default value is `none`.
+codec | Yes | Codec | The codec to apply. Must be `newline`, `json`, or `csv`. 
 sqs | Yes | sqs | The [Amazon Simple Queue Service](https://aws.amazon.com/sqs/) (Amazon SQS) configuration. See [sqs](#sqs) for details.
 aws | Yes | aws | The AWS configuration. See [aws](#aws) for details.
-on_error | No | String |  Determines how to handle errors in Amazon SQS. Can be either `retain_messages` or `delete_messages`. If `retain_messages`, then Data Prepper will leave the message in the SQS queue and try again. This is recommended for dead-letter queues. If `delete_messages`, then Data Prepper will delete failed messages. Default is `retain_messages`.
-buffer_timeout | No | Duration | The timeout for writing events to the Data Prepper buffer. Any events that the S3Source cannot write to the buffer in this time will be discarded. Default is 10 seconds.
-records_to_accumulate | No | Integer | The number of messages that accumulate before writing to the buffer. Default is 100.
+on_error | No | String |  Determines how to handle errors in Amazon SQS. Can be either `retain_messages` or `delete_messages`. If `retain_messages`, then Data Prepper will leave the message in the SQS queue and try again. This is recommended for dead-letter queues. If `delete_messages`, then Data Prepper will delete failed messages. Default value is `retain_messages`.
+buffer_timeout | No | Duration | The timeout for writing events to the Data Prepper buffer. Any events that the S3Source cannot write to the buffer in this time will be discarded. Default value is 10 seconds.
```

Comment:
@asifsmohammed Can you confirm that "The amount of time allowed for" is appropriate?

Reply:
I think timeout looks appropriate for me from a technical perspective. "Amount of time" sounds good to me as long as you mention that it will timeout after that time. I think if timeout is not used it's not clear what will happen after the time allowed is exceeded.

For example, (This is taken from grok processor `timeout_millis`)
An `int` that specifies the maximum amount of time, in milliseconds, that matching will be performed on an individual Record before it times out and moves on to the next Record.